File: frontend/src/api/priceCatalog.ts
import request from './request'

export interface PriceCatalogItem {
  id: string
  country: string | null
  name: string
  cert_type: string | null
  sample_qty: number | null
  based_on_report: string | null
  lead_weeks: number | null
  includes_testing: string | null
  cert_validity_years: number | null
  series_apply: string | null
  ref_price: number | null
  remark: string | null
  created_at: string
  updated_at: string
}

export interface PriceCatalogSearchItem {
  id: string
  country: string | null
  name: string
  based_on_report: string | null
  lead_weeks: number | null
  includes_testing: string | null
  series_apply: string | null
  ref_price: number | null
}

export const priceCatalogApi = {
  list(params: { page?: number; page_size?: number; keyword?: string }) {
    return request.get<any, { code: number; data: PriceCatalogItem[]; total: number }>('/price-catalog', { params })
  },
  search(params: { keyword?: string; country?: string }) {
    return request.get<any, { code: number; data: PriceCatalogSearchItem[] }>('/price-catalog/search', { params })
  },
  countries() {
    return request.get<any, { code: number; data: string[] }>('/price-catalog/countries')
  },
  create(data: Record<string, unknown>) {
    return request.post<any, { code: number; data: PriceCatalogItem }>('/price-catalog', data)
  },
  update(id: string, data: Record<string, unknown>) {
    return request.put<any, { code: number; data: PriceCatalogItem }>(`/price-catalog/${id}`, data)
  },
  delete(id: string) {
    return request.delete(`/price-catalog/${id}`)
  },
}

File: backend/app/schemas/price_catalog.py
import uuid
from datetime import datetime
from decimal import Decimal

from pydantic import BaseModel, ConfigDict


class PriceCatalogBase(BaseModel):
    country: str | None = None
    name: str
    cert_type: str | None = None
    sample_qty: int | None = None
    based_on_report: str | None = None
    lead_weeks: int | None = None
    includes_testing: str | None = None
    cert_validity_years: int | None = None
    series_apply: str | None = None
    ref_price: Decimal | None = None
    remark: str | None = None


class PriceCatalogCreate(PriceCatalogBase):
    pass


class PriceCatalogUpdate(BaseModel):
    country: str | None = None
    name: str | None = None
    cert_type: str | None = None
    sample_qty: int | None = None
    based_on_report: str | None = None
    lead_weeks: int | None = None
    includes_testing: str | None = None
    cert_validity_years: int | None = None
    series_apply: str | None = None
    ref_price: Decimal | None = None
    remark: str | None = None


class PriceCatalogOut(PriceCatalogBase):
    model_config = ConfigDict(from_attributes=True)
    id: uuid.UUID
    created_at: datetime
    updated_at: datetime


class PriceCatalogSearchItem(BaseModel):
    """报价明细搜索下拉用，返回所有可带入字段"""
    model_config = ConfigDict(from_attributes=True)
    id: uuid.UUID
    country: str | None
    name: str
    based_on_report: str | None
    lead_weeks: int | None
    includes_testing: str | None
    series_apply: str | None
    ref_price: Decimal | None
